src/luxonis_ml/data/utils/data_utils.py

```diff
@@ -102,11 +48,22 @@ def check_classification(val1, val2):
 
 
 def check_boxes(dataset, val1, val2):
-    if val2 is None and val1 is not None:
+    if (val1 is None and val2 is not None) or (val2 is None and val1 is not None):
         return [{"boxes": val1}]
+    elif val1 is None and val2 is None:
+        return []
 
     if len(val1) == len(val2["detections"]):
         for val1, val2 in list(zip(val1, val2["detections"])):
+            # assert bounding boxes contain the right format of either int and list or str and list
+            if not (
```

Comment:
Should we maybe also add x1,y1,w,h check? I assumed it should be in Yolo format and then training was failing. I bet I won't be the only one so perhaps validating this here would still be worth it.

Reply:
Can we actually distinguish between xywh and xyxy? Is that the check you mean? 

Or do you mean checking for float and 0-1 range or something like that?

Reply:
> Or do you mean checking for float and 0-1 range or something like that?

Yeah, and for x1,y1,w,h I meant just some basic checks. Like we know x1+w < 1, where as if x1 and x2 are passed it could happen it's higher. Like we can't check all, but if there is such scenario then it's preventable basically.